File: HFSGuardaDiretorio_SharpDevelop_Python/gui/FrmInfoDiretorio.py
 
from System import *
from System.Drawing import *
from System.Windows.Forms import *
from HFSGuardaDiretorio.objetos import *

class FrmInfoDiretorio(Form):
	def __init__(self):
		self._components = None
		self.InitializeComponent()
		self._fonte = Font(lvInfo.Items[0].Font, FontStyle.Bold)

	def InitializeComponent(self):
		listViewItem11 = System.Windows.Forms.ListViewItem("Aba:")
		listViewItem12 = System.Windows.Forms.ListViewItem("Nome:")
		listViewItem13 = System.Windows.Forms.ListViewItem("Tamanho:")
		listViewItem14 = System.Windows.Forms.ListViewItem("Tipo:")
		listViewItem15 = System.Windows.Forms.ListViewItem("Modificado:")
		listViewItem16 = System.Windows.Forms.ListViewItem("Atributos:")
		listViewItem17 = System.Windows.Forms.ListViewItem("Caminho:")
		self._groupBox1 = System.Windows.Forms.GroupBox()
		self._label5 = System.Windows.Forms.Label()
		self._label6 = System.Windows.Forms.Label()
		self._label3 = System.Windows.Forms.Label()
		self._label4 = System.Windows.Forms.Label()
		self._label2 = System.Windows.Forms.Label()
		self._label1 = System.Windows.Forms.Label()
		self._lvInfo = System.Windows.Forms.ListView()
		self._columnHeader1 = System.Windows.Forms.ColumnHeader()
		self._columnHeader2 = System.Windows.Forms.ColumnHeader()
		self._btnOk = System.Windows.Forms.Button()
		self._groupBox1.SuspendLayout()
		self.SuspendLayout()
		# 
		# groupBox1
		# 
		self._groupBox1.Controls.Add(self._label5)
		self._groupBox1.Controls.Add(self._label6)
		self._groupBox1.Controls.Add(self._label3)
		self._groupBox1.Controls.Add(self._label4)
		self._groupBox1.Controls.Add(self._label2)
		self._groupBox1.Controls.Add(self._label1)
		self._groupBox1.Location = System.Drawing.Point(12, 12)
		self._groupBox1.Name = "groupBox1"
		self._groupBox1.Size = System.Drawing.Size(340, 96)
		self._groupBox1.TabIndex = 0
		self._groupBox1.TabStop = False
		self._groupBox1.Text = "Legenda dos Atributos"
		# 
		# label5
		# 
		self._label5.Location = System.Drawing.Point(166, 62)
		self._label5.Name = "label5"
		self._label5.Size = System.Drawing.Size(168, 23)
		self._label5.TabIndex = 5
		self._label5.Text = "[ROL] - Arquivo Somente Leitura"
		self._label5.TextAlign = System.Drawing.ContentAlignment.MiddleLeft
		# 
		# label6
		# 
		self._label6.Location = System.Drawing.Point(6, 62)
		self._label6.Name = "label6"
		self._label6.Size = System.Drawing.Size(154, 23)
		self._label6.TabIndex = 4
		self._label6.Text = "[SYS] - Arquivo de Sistema"
		self._label6.TextAlign = System.Drawing.ContentAlignment.MiddleLeft
		# 
		# label3
		# 
		self._label3.Location = System.Drawing.Point(166, 39)
		self._label3.Name = "label3"
		self._label3.Size = System.Drawing.Size(168, 23)
		self._label3.TabIndex = 3
		self._label3.Text = "[VOL] - Volume ID"
		self._label3.TextAlign = System.Drawing.ContentAlignment.MiddleLeft
		# 
		# label4
		# 
		self._label4.Location = System.Drawing.Point(6, 39)
		self._label4.Name = "label4"
		self._label4.Size = System.Drawing.Size(154, 23)
		self._label4.TabIndex = 2
		self._label4.Text = "[HID] - Arquivo Oculto"
		self._label4.TextAlign = System.Drawing.ContentAlignment.MiddleLeft
		# 
		# label2
		# 
		self._label2.Location = System.Drawing.Point(166, 16)
		self._label2.Name = "label2"
		self._label2.Size = System.Drawing.Size(168, 23)
		self._label2.TabIndex = 1
		self._label2.Text = "[DIR] - Diretório"
		self._label2.TextAlign = System.Drawing.ContentAlignment.MiddleLeft
		# 
		# label1
		# 
		self._label1.Location = System.Drawing.Point(6, 16)
		self._label1.Name = "label1"
		self._label1.Size = System.Drawing.Size(154, 23)
		self._label1.TabIndex = 0
		self._label1.Text = "[ARQ] - Arquivo comum"
		self._label1.TextAlign = System.Drawing.ContentAlignment.MiddleLeft
		# 
		# lvInfo
		# 
		self._lvInfo.Columns.AddRange(System.Array[System.Windows.Forms.ColumnHeader](
			[self._columnHeader1,
			self._columnHeader2]))
		self._lvInfo.GridLines = True
		self._lvInfo.Items.AddRange(System.Array[System.Windows.Forms.ListViewItem](
			[listViewItem11,
			listViewItem12,
			listViewItem13,
			listViewItem14,
			listViewItem15,
			listViewItem16,
			listViewItem17]))
		self._lvInfo.Location = System.Drawing.Point(18, 114)
		self._lvInfo.MultiSelect = False
		self._lvInfo.Name = "lvInfo"
		self._lvInfo.OwnerDraw = True
		self._lvInfo.Size = System.Drawing.Size(328, 181)
		self._lvInfo.TabIndex = 1
		self._lvInfo.UseCompatibleStateImageBehavior = False
		self._lvInfo.View = System.Windows.Forms.View.Details
		self._lvInfo.DrawColumnHeader += self.LvInfoDrawColumnHeader
		self._lvInfo.DrawItem += self.LvInfoDrawItem
		self._lvInfo.DrawSubItem += self.LvInfoDrawSubItem
		# 
		# columnHeader1
		# 
		self._columnHeader1.Text = "Item"
		self._columnHeader1.Width = 108
		# 
		# columnHeader2
		# 
		self._columnHeader2.Text = "Descrição"
		self._columnHeader2.Width = 214
		# 
		# btnOk
		# 
		self._btnOk.Location = System.Drawing.Point(146, 309)
		self._btnOk.Name = "btnOk"
		self._btnOk.Size = System.Drawing.Size(75, 23)
		self._btnOk.TabIndex = 2
		self._btnOk.Text = "&Ok"
		self._btnOk.UseVisualStyleBackColor = True
		self._btnOk.Click += self.BtnOkClick
		# 
		# FrmInfoDiretorio
		# 
		self.AutoScaleDimensions = System.Drawing.SizeF(6, 13)
		self.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font
		self.ClientSize = System.Drawing.Size(363, 344)
		self.Controls.Add(self._btnOk)
		self.Controls.Add(self._lvInfo)
		self.Controls.Add(self._groupBox1)
		self.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
		self.MaximizeBox = False
		self.MinimizeBox = False
		self.Name = "FrmInfoDiretorio"
		self.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent
		self.Text = "Informações do Diretório / Arquivo"
		self._groupBox1.ResumeLayout(False)
		self.ResumeLayout(False)

	def Dispose(self, disposing):
		if disposing:
			if self._components != None:
				self._components.Dispose()
		self.Dispose(disposing)

	def BtnOkClick(self, sender, e):
		self.Close()

	def setDiretorio(self, diretorio):
		if diretorio != None:
			lvInfo.Items[0].SubItems.Add(diretorio.Aba.Nome)
			lvInfo.Items[1].SubItems.Add(diretorio.Nome)
			lvInfo.Items[2].SubItems.Add(diretorio.TamanhoFormatado)
			lvInfo.Items[3].SubItems.Add(diretorio.Tipo.Nome)
			lvInfo.Items[4].SubItems.Add(diretorio.ModificadoFormatado)
			lvInfo.Items[5].SubItems.Add(diretorio.Atributos)
			lvInfo.Items[6].SubItems.Add(diretorio.Caminho)
			lvInfo.Columns[1].Width = diretorio.Caminho.Length * 8

	def LvInfoDrawColumnHeader(self, sender, e):
		e.DrawDefault = True

	def LvInfoDrawItem(self, sender, e):
		e.DrawBackground()
		if (e.State & ListViewItemStates.Selected) == ListViewItemStates.Selected:
			e.Graphics.FillRectangle(SystemBrushes.Highlight, e.Bounds)
			e.Item.ForeColor = SystemColors.HighlightText
		else:
			e.Item.ForeColor = SystemColors.WindowText
		e.Item.Font = self._fonte
		e.DrawText()

	def LvInfoDrawSubItem(self, sender, e):
		e.DrawBackground()
		if (e.ItemState & ListViewItemStates.Selected) == ListViewItemStates.Selected:
			e.Graphics.FillRectangle(SystemBrushes.Highlight, e.Bounds)
			e.SubItem.ForeColor = SystemColors.HighlightText
		else:
			e.SubItem.ForeColor = SystemColors.WindowText
		e.DrawText()

File: HFSGuardaDiretorio_SharpDevelop_Ruby/gui/FrmInfoDiretorio.rb
# 
# * Created by SharpDevelop.
# * User: henrique.souza
# * Date: 11/12/2014
# * Time: 11:11
# *
# * To change this template use Tools | Options | Coding | Edit Standard Headers.
# 
require "mscorlib"
require "System, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Drawing, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "HFSGuardaDiretorio.objetos, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"

module HFSGuardaDiretorio.gui

	class FrmInfoDiretorio < System::Windows::Forms::Form
		def initialize()
			@components = nil
			self.InitializeComponent()
			@fonte = Font.new(lvInfo.Items[0].Font, FontStyle.Bold)
		end

		def InitializeComponent()
			listViewItem1 = System::Windows::Forms::ListViewItem.new("Aba:")
			listViewItem2 = System::Windows::Forms::ListViewItem.new("Nome:")
			listViewItem3 = System::Windows::Forms::ListViewItem.new("Tamanho:")
			listViewItem4 = System::Windows::Forms::ListViewItem.new("Tipo:")
			listViewItem5 = System::Windows::Forms::ListViewItem.new("Modificado:")
			listViewItem6 = System::Windows::Forms::ListViewItem.new("Atributos:")
			listViewItem7 = System::Windows::Forms::ListViewItem.new("Caminho:")
			@groupBox1 = System::Windows::Forms::GroupBox.new()
			@label5 = System::Windows::Forms::Label.new()
			@label6 = System::Windows::Forms::Label.new()
			@label3 = System::Windows::Forms::Label.new()
			@label4 = System::Windows::Forms::Label.new()
			@label2 = System::Windows::Forms::Label.new()
			@label1 = System::Windows::Forms::Label.new()
			@lvInfo = System::Windows::Forms::ListView.new()
			@columnHeader1 = System::Windows::Forms::ColumnHeader.new()
			@columnHeader2 = System::Windows::Forms::ColumnHeader.new()
			@btnOk = System::Windows::Forms::Button.new()
			@groupBox1.SuspendLayout()
			self.SuspendLayout()
			# 
			# groupBox1
			# 
			@groupBox1.Controls.Add(@label5)
			@groupBox1.Controls.Add(@label6)
			@groupBox1.Controls.Add(@label3)
			@groupBox1.Controls.Add(@label4)
			@groupBox1.Controls.Add(@label2)
			@groupBox1.Controls.Add(@label1)
			@groupBox1.Location = System::Drawing::Point.new(12, 12)
			@groupBox1.Name = "groupBox1"
			@groupBox1.Size = System::Drawing::Size.new(340, 96)
			@groupBox1.TabIndex = 0
			@groupBox1.TabStop = false
			@groupBox1.Text = "Legenda dos Atributos"
			# 
			# label5
			# 
			@label5.Location = System::Drawing::Point.new(166, 62)
			@label5.Name = "label5"
			@label5.Size = System::Drawing::Size.new(168, 23)
			@label5.TabIndex = 5
			@label5.Text = "[ROL] - Arquivo Somente Leitura"
			@label5.TextAlign = System::Drawing::ContentAlignment.MiddleLeft
			# 
			# label6
			# 
			@label6.Location = System::Drawing::Point.new(6, 62)
			@label6.Name = "label6"
			@label6.Size = System::Drawing::Size.new(154, 23)
			@label6.TabIndex = 4
			@label6.Text = "[SYS] - Arquivo de Sistema"
			@label6.TextAlign = System::Drawing::ContentAlignment.MiddleLeft
			# 
			# label3
			# 
			@label3.Location = System::Drawing::Point.new(166, 39)
			@label3.Name = "label3"
			@label3.Size = System::Drawing::Size.new(168, 23)
			@label3.TabIndex = 3
			@label3.Text = "[VOL] - Volume ID"
			@label3.TextAlign = System::Drawing::ContentAlignment.MiddleLeft
			# 
			# label4
			# 
			@label4.Location = System::Drawing::Point.new(6, 39)
			@label4.Name = "label4"
			@label4.Size = System::Drawing::Size.new(154, 23)
			@label4.TabIndex = 2
			@label4.Text = "[HID] - Arquivo Oculto"
			@label4.TextAlign = System::Drawing::ContentAlignment.MiddleLeft
			# 
			# label2
			# 
			@label2.Location = System::Drawing::Point.new(166, 16)
			@label2.Name = "label2"
			@label2.Size = System::Drawing::Size.new(168, 23)
			@label2.TabIndex = 1
			@label2.Text = "[DIR] - Diretório"
			@label2.TextAlign = System::Drawing::ContentAlignment.MiddleLeft
			# 
			# label1
			# 
			@label1.Location = System::Drawing::Point.new(6, 16)
			@label1.Name = "label1"
			@label1.Size = System::Drawing::Size.new(154, 23)
			@label1.TabIndex = 0
			@label1.Text = "[ARQ] - Arquivo comum"
			@label1.TextAlign = System::Drawing::ContentAlignment.MiddleLeft
			# 
			# lvInfo
			# 
			@lvInfo.Columns.AddRange(System::Array[System::Windows::Forms::ColumnHeader].new([@columnHeader1, @columnHeader2]))
			@lvInfo.GridLines = true
			@lvInfo.Items.AddRange(System::Array[System::Windows::Forms::ListViewItem].new([listViewItem1, listViewItem2, listViewItem3, listViewItem4, listViewItem5, listViewItem6, listViewItem7]))
			@lvInfo.Location = System::Drawing::Point.new(18, 114)
			@lvInfo.MultiSelect = false
			@lvInfo.Name = "lvInfo"
			@lvInfo.OwnerDraw = true
			@lvInfo.Size = System::Drawing::Size.new(328, 181)
			@lvInfo.TabIndex = 1
			@lvInfo.UseCompatibleStateImageBehavior = false
			@lvInfo.View = System::Windows::Forms::View.Details
			@lvInfo.DrawColumnHeader { |sender, e| @LvInfoDrawColumnHeader(sender, e) }
			@lvInfo.DrawItem { |sender, e| @LvInfoDrawItem(sender, e) }
			@lvInfo.DrawSubItem { |sender, e| @LvInfoDrawSubItem(sender, e) }
			# 
			# columnHeader1
			# 
			@columnHeader1.Text = "Item"
			@columnHeader1.Width = 108
			# 
			# columnHeader2
			# 
			@columnHeader2.Text = "Descrição"
			@columnHeader2.Width = 214
			# 
			# btnOk
			# 
			@btnOk.Location = System::Drawing::Point.new(146, 309)
			@btnOk.Name = "btnOk"
			@btnOk.Size = System::Drawing::Size.new(75, 23)
			@btnOk.TabIndex = 2
			@btnOk.Text = "&Ok"
			@btnOk.UseVisualStyleBackColor = true
			@btnOk.Click { |sender, e| @BtnOkClick(sender, e) }
			# 
			# FrmInfoDiretorio
			# 
			self.AutoScaleDimensions = System::Drawing::SizeF.new(6, 13)
			self.AutoScaleMode = System::Windows::Forms::AutoScaleMode.Font
			self.ClientSize = System::Drawing::Size.new(363, 344)
			self.Controls.Add(@btnOk)
			self.Controls.Add(@lvInfo)
			self.Controls.Add(@groupBox1)
			self.FormBorderStyle = System::Windows::Forms::FormBorderStyle.FixedDialog
			self.MaximizeBox = false
			self.MinimizeBox = false
			self.Name = "FrmInfoDiretorio"
			self.StartPosition = System::Windows::Forms::FormStartPosition.CenterParent
			self.Text = "Informações do Diretório / Arquivo"
			@groupBox1.ResumeLayout(false)
			self.ResumeLayout(false)
		end

		def Dispose(disposing)
			if disposing then
				if @components != nil then
					@components.Dispose()
				end
			end
			self.Dispose(disposing)
		end

		def BtnOkClick(sender, e)
			self.Close()
		end

		def setDiretorio(diretorio)
			if diretorio != nil then
				lvInfo.Items[0].SubItems.Add(diretorio.Aba.Nome)
				lvInfo.Items[1].SubItems.Add(diretorio.Nome)
				lvInfo.Items[2].SubItems.Add(diretorio.TamanhoFormatado)
				lvInfo.Items[3].SubItems.Add(diretorio.Tipo.Nome)
				lvInfo.Items[4].SubItems.Add(diretorio.ModificadoFormatado)
				lvInfo.Items[5].SubItems.Add(diretorio.Atributos)
				lvInfo.Items[6].SubItems.Add(diretorio.Caminho)
				lvInfo.Columns[1].Width = diretorio.Caminho.Length * 8
			end
		end

		def LvInfoDrawColumnHeader(sender, e)
			e.DrawDefault = true
		end

		def LvInfoDrawItem(sender, e)
			e.DrawBackground()
			if (e.State & ListViewItemStates.Selected) == ListViewItemStates.Selected then
				e.Graphics.FillRectangle(SystemBrushes.Highlight, e.Bounds)
				e.Item.ForeColor = SystemColors.HighlightText
			else
				e.Item.ForeColor = SystemColors.WindowText
			end
			e.Item.Font = @fonte
			e.DrawText()
		end

		def LvInfoDrawSubItem(sender, e)
			e.DrawBackground()
			if (e.ItemState & ListViewItemStates.Selected) == ListViewItemStates.Selected then
				e.Graphics.FillRectangle(SystemBrushes.Highlight, e.Bounds)
				e.SubItem.ForeColor = SystemColors.HighlightText
			else
				e.SubItem.ForeColor = SystemColors.WindowText
			end
			e.DrawText()
		end
	end
end